confidential/confidential.go

```diff
@@ -91,3 +91,34 @@ func FinalValueBlindingFactor(input FinalValueBlindingFactorInput) ([32]byte, er
 
 	return secp256k1.BlindGeneratorBlindSum(ctx, values, generatorBlind, blindingFactor, len(input.InValues))
 }
+
+//AssetCommitment method generates asset commitment
+func AssetCommitment(asset []byte, factor []byte) ([33]byte, error) {
+	ctx, _ := secp256k1.ContextCreate(secp256k1.ContextBoth)
+	defer secp256k1.ContextDestroy(ctx)
+
+	generator, err := secp256k1.GeneratorGenerateBlinded(ctx, asset, factor)
+	if err != nil {
+		return [33]byte{}, err
+	}
+
+	return secp256k1.GeneratorSerialize(ctx, generator), nil
```

Comment:
You should be able to call the `Bytes` method of the `*Generator` type instead of calling the GeneratorSerialize binding:
```suggestion
	return generator.Bytes(), nil
```

Reply:
Done